### PR TITLE
Make sure that organized point clouds are still organized after transformPointCloud

### PR DESCRIPTION
**Pull request by mvieth**

### CODE REVIEW

File: common/include/pcl/common/impl/transforms.hpp
Comment (larshg) [inline common/include/pcl/common/impl/transforms.hpp:322]:
```suggestion
    cloud_out.reserve (cloud_in.size ());
```

**Comment by larshg** [comment]
Is it resize that sets width to count and heigt to 1?

Which you then set properly afterwards?

**Comment by larshg** [comment]
If the cloud contains NaN or other non-finite points, they are filtered out, which makes it non-organized?

**Comment by kunaltyagi** [comment]
I'm updating a PR that allows the intention to be set in a better manner.

PS: resize doesn't touch width or height if the new size == width * height

**Comment by mvieth** [comment]
> Is it resize that sets width to count and heigt to 1?
> Which you then set properly afterwards?

No, it is [`assign`](https://github.com/PointCloudLibrary/pcl/blob/e5f4a176ed148a0e2dad169d94a016c344eacd9f/common/include/pcl/point_cloud.h#L518).

> If the cloud contains NaN or other non-finite points, they are filtered out, which makes it non-organized?

They are not really filtered out, they just not set in `cloud_out`.

**Review by larshg — APPROVED** [review]
Ah, thanks for the explanation.

**Comment by mvieth** [comment]
> I'm updating a PR that allows the intention to be set in a better manner (#4397)
> 
> PS: resize doesn't touch width or height if the new size == width * height

I will try to look at that PR soon, I am a bit slow with reviews currently, sorry.
Your PR and this one don't clash in any way, do they?

**Comment by kunaltyagi** [comment]
There's no clash, just that I added method for assign that takes 2 parameters (width and height), so using that PR to resolve the issue would make the code a bit cleaner

**Comment by mvieth** [comment]
> There's no clash, just that I added method for assign that takes 2 parameters (width and height), so using that PR to resolve the issue would make the code a bit cleaner

@kunaltyagi Do you absolutely want to wait until the other PR is merged, or should we move forward with this PR in its current state?

**Comment by kunaltyagi** [comment]
Sorry, I didn't realize that PR was stale. It needed a tiny update to get the CI green.

My reason for waiting on this PR is that refactoring later on is difficult, since there's plenty of refactoring to go around already

**Review by kunaltyagi — APPROVED** [review]

**Pull request merged**
